### ISSUE TITLE
Preprocess Data

### ISSUE THREAD

**Issue by Dhakshitha7** [Dhakshitha7/Bharat-Finance-Credit-Risk-Modeling#5]
# Encode categorical features
for col in df.select_dtypes(include='object').columns:
    if df[col].nunique() <= 2:
        df[col] = LabelEncoder().fit_transform(df[col])
    else:
        df = pd.get_dummies(df, columns=[col], drop_first=True)

# Fill missing values
df.fillna(df.median(numeric_only=True), inplace=True)

# Features and label
X = df.drop("Churn", axis=1)
y = df["Churn"]
